src/app/util/IcdMonitoringTable.h

```diff
@@ -0,0 +1,80 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <lib/core/CHIPConfig.h>
+#include <lib/core/CHIPPersistentStorageDelegate.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/PersistentData.h>
+#include <stddef.h>
+
+namespace chip {
+
+constexpr size_t kIcdMonitoringBufferSize = 1024;
+
+struct IcdMonitoringEntry : public PersistentData<kIcdMonitoringBufferSize>,
+                            chip::app::Clusters::IcdManagement::Structs::MonitoringRegistrationStruct::Type
+{
+    IcdMonitoringEntry(FabricIndex fabric = kUndefinedFabricIndex, NodeId nodeId = kUndefinedNodeId)
+    {
+        this->fabricIndex      = fabric;
+        this->checkInNodeID    = nodeId;
+        this->monitoredSubject = nodeId;
+    }
+    bool IsValid() { return this->checkInNodeID != kUndefinedNodeId && this->fabricIndex != kUndefinedFabricIndex; }
+
+    CHIP_ERROR UpdateKey(StorageKeyName & key) override;
+    CHIP_ERROR Serialize(TLV::TLVWriter & writer) const override;
```

Comment:
I don't understand how the key bits work.  What points the key span to the right buffer?

Reply:
The buffer used to be allocated in the stack of the Serialize/Deserialize methods, becoming invalid upon return. Now the buffer belongs to the PersistentData entry object, so the key is valid as long as the IcdMonitoringEntry instance that contains it.

Reply:
I think the real answer here is: The TLVReader points to our internal buffer (from PersistentData bits), so the Span ends up pointing into that buffer.  Right?